File: core/src/com/spacefist/ai/defensiveai/DefendState.java
package com.spacefist.ai.defensiveai;

import com.spacefist.ai.ShipEnemyInfo;
import com.spacefist.ai.ShipInfo;
import com.spacefist.ai.abst.EnemyAI;
import com.spacefist.ai.abst.EnemyAIState;
import com.spacefist.ai.abst.FuzzyLogicEnabled;
import com.spacefist.entities.enemies.Enemy;

/**
 * The fuzzy state for a non-firing AI
 */
public class DefendState extends FuzzyLogicEnabled implements EnemyAIState {
    private EnemyAI       ai;
    private ShipInfo      shipInfo;
    private ShipEnemyInfo shipEnemyInfo;
    private Enemy         enemy;

    public DefendState(EnemyAI ai) {
        this.ai       = ai;
        shipInfo      = ai.getShipInfo();
        shipEnemyInfo = ai.getShipEnemyInfo();
    }

    public void update() { }
}

File: core/src/com/spacefist/ai/defensiveai/DefendState.kt
package com.spacefist.ai.defensiveai

import com.spacefist.ai.ShipEnemyInfo
import com.spacefist.ai.ShipInfo
import com.spacefist.ai.abst.EnemyAI
import com.spacefist.ai.abst.EnemyAIState
import com.spacefist.ai.abst.FuzzyLogicEnabled
import com.spacefist.entities.enemies.Enemy

/**
 * The fuzzy state for a non-firing AI
 */
data class DefendState(private val ai: EnemyAI) : FuzzyLogicEnabled(), EnemyAIState {
    private val shipInfo:      ShipInfo      = ai.shipInfo
    private val shipEnemyInfo: ShipEnemyInfo = ai.shipEnemyInfo
    private val enemy:         Enemy?        = null

    override fun update() {}
}
